models.py

```diff
@@ -178,55 +178,67 @@ def empty_order(self):
-
-        :param item Item: instance of models.Item
+        Add items to the order from a dict {<Item>: <int:quantity>}. Handles
+        creating or updating the OrderItem cross table and the Order total
+        :param dict items: {<Item>: <quantity:int>}
         """
-
-        for orderitem in self.order_items:
-            # Looping all the OrderItem related to this order, if one with the
-            # same item is found we update that row.
-            if orderitem.item == item:
-                orderitem.add_item(quantity)
+        with database.atomic():
+            for (item, quantity) in items.items():
+                for orderitem in self.order_items:
+                    # Looping all the OrderItem related to this order,
+                    # if one with the same item is found we update that row.
+                    if orderitem.item == item:
+                        orderitem.add_item(quantity)
+                        continue
+
+                # if no matching existing OrderItem is found
+                # create a new row in the OrderItem table
+                OrderItem.create(
+                    order=self,
+                    item=item,
+                    quantity=quantity,
+                    subtotal=item.price * quantity
+                )
 
                 self.total_price += (item.price * quantity)
-                self.save()
-                return self
-
-        # if no existing OrderItem is found with this order and this Item,
-        # create a new row in the OrderItem table
-        OrderItem.create(
-            order=self,
-            item=item,
-            quantity=quantity,
-            subtotal=item.price * quantity
-        )
+            self.save()
+        return self
 
-        self.total_price += (item.price * quantity)
-        self.save()
+    def remove_items(self, items):
+        """
+        Remove items from an order, handling the relative OrderItem row and
+        the Order total price update.
+        :param dict items: {<Item>: <quantity:int>}
+        """
+        with database.atomic():
+            for item, quantity in items.items():
+                for orderitem in self.order_items:
+                    if orderitem.item == item:
+                        removed_items = orderitem.remove_item(quantity)
+                        self.total_price -= (item.price * removed_items)
+                    else:
+                        pass
+                        # NOTE: We should do something when the item is not
+                        # found
+            self.save()
         return self
 
-    def remove_item(self, item, quantity=1):
+    def add_item(self, item, quantity=1):
         """
-        Remove the given item from the order, reducing quantity of the relative
-        OrderItem entity or deleting it if removing the last item
-        (OrderItem.quantity == 0)
+        Add one item to the order calling `add_items`.
```

Comment:
Questa docstring non mi piace. La docstring è per chi utilizza il metodo, e tipicamente non riguarda (o riguarda in maniera secondaria) gli internals del metodo. Se vuoi quindi dire che è solo una shortcut per l'add_items va bene, ma deve rimanere la parte dove si spiega come va usata.